packages/esm-ward-app/src/ward-workspace/ward-patient-notes/form/notes-form.component.tsx

```diff
@@ -0,0 +1,171 @@
+
+const noteFormSchema = z.object({
+  wardClinicalNote: z.string().refine((val) => val.trim().length > 0, {
+    //t('clinicalNoteErrorMessage','Clinical note is required')
+    message: translateFrom(moduleName, 'clinicalNoteErrorMessage', 'Clinical note is required'),
+  }),
+});
+
+interface PatientNotesFormProps extends DefaultPatientWorkspaceProps {
+  onWorkspaceClose: () => void;
+}
+
+const PatientNotesForm: React.FC<PatientNotesFormProps> = ({
+  closeWorkspaceWithSavedChanges,
+  patientUuid,
+  promptBeforeClosing,
+  onWorkspaceClose,
+}) => {
+  const { emrConfiguration, isLoadingEmrConfiguration, errorFetchingEmrConfiguration } = useEmrConfiguration();
+  const { t } = useTranslation();
+  const session = useSession();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [rows, setRows] = useState<number>();
+
+  const { control, handleSubmit, watch, getValues, setValue, formState } = useForm<NotesFormData>({
+    mode: 'onSubmit',
+    resolver: zodResolver(noteFormSchema),
+    defaultValues: {
+      wardClinicalNote: '',
+    },
+  });
+
+  const { isDirty } = formState;
+
+  useEffect(() => {
+    promptBeforeClosing(() => isDirty);
+  }, [isDirty, promptBeforeClosing]);
+
+  const locationUuid = session?.sessionLocation?.uuid;
+  const providerUuid = session?.currentProvider?.uuid;
+
+  const onSubmit = (data: NotesFormData) => {
+    const { wardClinicalNote } = data;
+    setIsSubmitting(true);
+
+    const notePayload = {
+      encounterDatetime: dayjs(new Date()).format(),
+      patient: patientUuid,
+      location: locationUuid,
+      encounterType: emrConfiguration?.visitNoteEncounterType.uuid,
+      encounterProviders: [
+        {
+          encounterRole: emrConfiguration?.clinicianEncounterRole.uuid,
+          provider: providerUuid,
+        },
+      ],
+      obs: wardClinicalNote
+        ? [
+            {
+              concept: { uuid: emrConfiguration?.consultFreeTextCommentsConcept.uuid, display: '' },
+              value: wardClinicalNote,
+            },
+          ]
+        : [],
+    };
+
+    const abortController = new AbortController();
+
+    savePatientNote(abortController, notePayload)
+      .then(() => {
+        closeWorkspaceWithSavedChanges({ onWorkspaceClose });
+        showSnackbar({
+          isLowContrast: true,
+          subtitle: t('patientNoteNowVisible', 'It should be now visible in the notes history'),
+          kind: 'success',
+          title: t('visitNoteSaved', 'Patient note saved'),
+        });
+      })
+      .catch((err) => {
+        createErrorHandler();
+
+        showSnackbar({
+          title: t('patientNoteSaveError', 'Error saving patient note'),
+          kind: 'error',
+          isLowContrast: false,
+          subtitle: err?.message,
+        });
+      })
+      .finally(() => {
+        setIsSubmitting(false);
+      });
+  };
+
+  const onError = (errors) => console.error(errors);
```

Comment:
On error, we should display all the errors as inline notifications at the end of the form.
The `errors` is not of instance `Error` but an object containing the field name as key and **error messages** as values.

Reply:
These are validation errors, that are handled by the `invalid` and `invalidText` attributes on the form fields